sklearn/inspection/_permutation_importance.py

```diff
@@ -116,9 +116,11 @@ def permutation_importance(estimator, X, y, scoring=None, n_repeats=5,
     baseline_score = scorer(estimator, X, y)
     scores = np.zeros((X.shape[1], n_repeats))
 
-    scores = Parallel(n_jobs=n_jobs)(delayed(_calculate_permutation_scores)(
-        estimator, X, y, col_idx, random_state, n_repeats, scorer
-    ) for col_idx in range(X.shape[1]))
+    max_nbytes = None if hasattr(X, 'loc') else '1M'
+    scores = Parallel(n_jobs=n_jobs, max_nbytes=max_nbytes)(
+        delayed(_calculate_permutation_scores)(
+            estimator, X, y, col_idx, random_state, n_repeats, scorer
+        ) for col_idx in range(X.shape[1]))
```

Comment:
we should pass an integer seed instead of a single `random_state` instance.

```
# Precompute a random seed from the random state to be used
# to get a fresh independent RandomState instance for each
# parallel call to _calculate_permutation_scores, irrespective of
# the fact that variables are shared or not depending on the active 
# joblib backend (sequential, thread-based or process-based).
random_state = check_random_state(random_state)
random_seed = random_state.randint(0, np.iinfo(np.int32).max)

scorer = check_scoring(estimator, scoring=scoring)
baseline_score = scorer(estimator, X, y)
scores = np.zeros((X.shape[1], n_repeats))
max_nbytes = None if hasattr(X, 'loc') else '1M'
scores = Parallel(n_jobs=n_jobs, max_nbytes=max_nbytes)(
    delayed(_calculate_permutation_scores)(
       estimator, X, y, col_idx, random_seed, n_repeats, scorer
    ) for col_idx in range(X.shape[1]))
```

Reply:
implemented in https://github.com/scikit-learn/scikit-learn/pull/15933

We therefore only need the additional test